### PR TITLE
Add PR builder GitHub Workflow for Code snippets

### PR DESCRIPTION
**Pull request by kwart**
This PR adds GitHub workflow to compile the Java code on pull requests (to verify the code still works with given Hazelcast version).
It also adds a dependabot configuration.

Originally the changes were part of the #1263

### CODE REVIEW

File: .github/workflows/maven-pr-builder.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          java-version: '17'
```
Comment (JackPGreen) [inline .github/workflows/maven-pr-builder.yaml:23]: [I think this should by dynamic](https://github.com/hazelcast/hz-docs/pull/1263#discussion_r1750302258).

**Review by JackPGreen — APPROVED** [review]

**Comment by kwart** [comment]
Thanks for the review, Jack.
I agree; we can try to move it even more in the next iterations. It just wasn't my goal in this PR.

**Review by oliverhowell — APPROVED** [review]
Great enhancement!
I've been talking about this recently and like magic, we have a PR!

**Comment by kwart** [comment]
Thank you for the review, Oliver.

**Pull request merged**